src/store/sagas/cloud.ts

```diff
@@ -87,7 +91,7 @@ function* cloudWorker( { payload } ) {
       const accounts: Accounts = yield select( state => state.accounts.accounts )
 
       const encKey = BHROperations.getDerivedKey(
-        bip39.mnemonicToSeedSync( wallet.primaryMnemonic ).toString( 'hex' ),
+        bip39.mnemonicToSeedSync( primaryMnemonic ).toString( 'hex' ),
```

Comment:
Improvement suggestion: Here instead of using primaryMnemonic from wallet, we can directly pick primarySeed from walletObj and pass it directly to BHROperations.getDerivedKey function(instead of first taking the primaryMnemonic and converting it into seed using mnemonicToSeedSync function).

Reply:
Resolved